APACHE_PROPOSAL.md

```diff
@@ -0,0 +1,326 @@
+8. `hugegraph-rpc`: https://github.com/hugegraph/hugegraph-rpc
+9. `hugegraph-hubble`: https://github.com/hugegraph/hugegraph-hubble
+10. `hugegraph-test`: https://github.com/hugegraph/hugegraph-test
+
+### Source and Intellectual Property Submission Plan
+
+The codes are currently under Apache License Version 2.0 and have been verified to there is no intellectual property or license issues before being released to open source by Baidu in 2018. Baidu will provide SGA and all committers will sign ICLA after HugeGraph is accepted into the Incubator.
+
+#### External Dependencies:
+
+As all dependencies are managed using Apache Maven, none of the external libraries need to be packaged in a source distribution. All dependencies have Apache compatible licenses except MySQL (GPL-2.0), we will remove MySQL dependencies in future.
+
+HugeGraph has the following external [dependencies](https://github.com/hugegraph/hugegraph/issues/1632):
+
+- Apache License
+
+  - HttpClient
+  - HttpCore
+
+- Apache License 2.0
+
+  - Annotations for Metrics
+  - ansj_seg
+  - Apache Cassandra
+  - Apache Commons
+  - Apache Groovy
+  - Apache HttpClient
+  - Apache HttpCore
+  - Apache Ivy
+  - Apache Log4j
+  - Apache Thrift
+  - Apache TinkerPop
+  - Apache Yetus
+  - Bean Validation API
+  - Caffeine cache
+  - CLI
+  - Sofa
+  - ConcurrentLinkedHashMap
+  - Data Mapper for Jackson
+  - DataStax Java Driver for Apache Cassandra
+  - Disruptor Framework
+  - error-prone annotations
+  - exp4j
+  - fastutil
+  - Findbugs
+  - Google Android Annotations Library
+  - Gson
+  - Guava
+  - Hibernate Validator Engine
+  - HPPC Collections
+  - htrace-core4
+  - IKAnalyzer
+  - io.grpc
+  - J2ObjC Annotations
+  - Jackson
+  - Java Agent for Memory Measurements
+  - Java Concurrency Tools Core Library
+  - JavaPoet
+  - javatuples
+  - jcseg-core
+  - jffi
+  - JJWT
+  - jnr
+  - Joda
+  - jraft
+  - JSON.simple
+  - JVM Integration for Metrics
+  - Lucene
+  - LZ4 and xxHash
+  - Metrics
+  - mmseg4j-core
+  - Netty
+  - Ning-compress-LZF
+  - nlp-lang
+  - Objenesis
+  - OHC core
+  - OpenTracing
+  - perfmark
+  - picocli
+  - proto-google-common-protos
+  - sigar
+  - snappy-java
+  - stream-lib
+  - swagger
+  - Thrift Server implementation backed by LMAX Disruptor
+  - jieba for java
+  - Java Native Access
+  - HanLP
+  - SnakeYAML
+
+- Apache License 2.0 + GPLv2 License
+
+  - RocksDB JNI
+
+- BSD License
+
+  - ANTLR 3
+  - ASM
+  - Chinese to Pinyin
+  - Hamcrest
+  - JLine
+  - jcabi
+  - PostgreSQL JDBC Driver - JDBC 4.2
+  - Protocol Buffers
+  - StringTemplate
+
+- CDDL + GPL License
+
+  - Grizzly
+  - Jersey
+
+- CDDL + GPLv2 License
+
+  - glassfish
+  - Java Servlet API
+  - javax.ws.rs-api 
+  - jersey-inject-hk2
+  - JSR 353 (JSON Processing) Default Provider
+
+- Commercial License + AGPL License
+
+  - TrueLicense
+
+- Eclipse Distribution License
+
+  - Eclipse Collections
+  - JUnit
+
+- ISC/BSD License
+
+  - jBCrypt
+
+- LGPL License
```

Comment:
We need to take a look at this dependency. We cannot ship the GPL and LGPL third party dependency.
https://www.apache.org/legal/resolved.html#category-x

Reply:
TODO: check LGPL dependency